### ISSUE TITLE
Reassessment of stimulation as schema, in techniques, and in protocols/behavioral protocols

### ISSUE THREAD

**Issue by lzehl** [openMetadataInitiative/openMINDS_core#346]
This issue is related to the discussion that started in [controlledTerms PR 293](https://github.com/HumanBrainProject/openMINDS_controlledTerms/pull/293)

Issue: we need a cleaner solution between stimulation as technique, stimulation concepts and stimulation as in-depth schema to describe exactly how a stimulation was applied.

The following lists represent the current status of stimulation and its occurrences in openMINDS:

**Stimulation Techniques in openMINDS:**
    • current step stimulation
    • checkerboard visual stimulation
    • drifting grating visual stimulation
    • electrical stimulation
    • figure-ground visual stimulation
    • Gestalt visual stimulation
    • microstimulation
    • natural image visual stimulation
    • natural sound auditory stimulation
    • optogenetic stimulation
    • photon stimulation
    • single pulse electrical stimulation
    • static grating visual stimulation
    • transcranial magnetic stimulation
    • whisker stimulation

**Behavioral protocol schema:**
    • described in (→ DOI/File/URL)
    • description
    • name
    • internal identifier
    • stimulation (→ **Stimulation**)

**Protocol schema:**
    • description
    • name
    • stimulation (→ **Stimulation**)
    • technique (→ Technique)

**Stimulation schema:**
    • description
    • lookup label
    • stimulation approach (→ StimulationApproach)
    • stimulus type (→ StimulusType)
    • custom property set (→ CustomPropertySet)
    • data location (→ File/FileBundle)

**StimulationApproaches in openMINDS:**
    • auditory stimulation
    • gustatory stimulation
    • interoceptive stimulation
    • olfactory stimulation
    • tactile stimulation
    • visual stimulation

**StimulusType in openMINDS:**
    • direct current
    • moving checkerboard
    • moving grating
    • moving symbol
    • natural movie
    • natural scene
    • static checkerboard
    • static grating
    • static symbol
    • textured surface
    • vibrating object
    • vibrating surface

**Comment by lzehl** [comment]
@Peyman-N and @apdavison please have a look at this issue and the idea below for our discussion on stimuli and stimulations:

@UlrikeS91 and I discussed this issue briefly and ended with the following idea: 
- We extract the stimulation techniques out of the Technique terminology and collect them separately in StimulationTechnique (new terminology). 
- The names/labels of these StimulationTechnique terms will always combine from stimulus type and stimulus approach (e.g., current step stimulation electrical stimulation, checkerboard visual stimulation). Generic approaches can be covered here as well (e.g., electrical stimulation, visual stimulation).
- We delete the Stimulation schema from core and remove also it linkage from BehavioralProtocol and Protocol. Instead we link from BehavioralProtocol and Protocol just to the new StimulationTechnique. 
- In the openMINDS_stimulation extension we should have at least one stimulation activity per stimulation approach (AuditoryStimulation, GustatoryStimulation, InteroceptiveStimulation, OlfactoryStimulation, TactileStimulation, VisualStimulation). These stimulation activities can be extended from a generic Stimulation schema. 
- I would assume that the generic Stimulation schema inherits from the Activity schema (or ExperimentalActivity schema).
- In addition we should think about schemas that in in-depth describe the parameters of the different stimuli (e.g., checkerboard stimulus schema that defines if it is moving or static, how many number of rows and columns, etc).

NOTE: the extension does not have to include all at once. We can start with the use cases we have. Also, please take the idea really as idea we can use to discuss further.

**Comment by lzehl** [comment]
meeting notes: @apdavison @UlrikeS91 @Peyman-N 

discussion results:
**1. change link from 'stimulation' property in BehavioralProtocol and Protocol to (Stimulation)Technique
2. separate stimulation techniques from other techniques (does not block v3.0.0 release but would be a nice to have)
3. delete Stimulation schema from core
4. in stimulation extension: capture stimulations as StimulationActivity with link to used Stimulus (there will be different Stimulus Schemas depending on the stimulation approach)
5. we need to have a RecordingActivity in parallel**

discussion points:
-  single stimulation vs a single stimulation sequence (e.g. of one trial) vs multiple (repeating) stimulation sequences (e.g. multiple trials): 
    - single stimulation: 
        - capture in one StimulationActivity with link to used Stimulus 
    - single stimulation sequence:
        - capture in multiple StimulationActivities each linking to one Stimulus ? (timing: start/end time property of activity) OR
        - capture in one StimulationActivity with ordered list of links to used Stimuli ? (timing: onset property with ordered QVs)
    - multiple stimulation sequences:
        - capture in one StimulationActivity with ordered list of links to all used Stimuli without defining a subsequence (timing of stimuli: onset property with ordered QVs)? OR
        - capture in multiple StimulationActivities each representing a (sub)sequence that defines a set of stimuli in ordered lists (timing of subsequences: start/end time of activity; timing of stimuli in each subsequence: onset property with ordered QVs (with relative times?)) OR
    -  maybe introduce a StimulusSet schema (with links to different single Stimulus with relative timings) that the StimulationActivity could link to in addition to a single Stimulus to solve sequence issue???

Note (from my side for making a decision):
I think we should try to minimize the documentation effort for capturing sequence repetitions. This could mean repetition of the same sequence for X times in one recording session or the alternation of a set of sequences (e.g., condition A,B,C in block or in random order with X repetitions for A,B,and C).

**Comment by lzehl** [comment]
@tgbugs please feel free to comment as well in this issue

**Comment by lzehl** [comment]
results of further discussion:
**+ make stimulus type a category and split terminologies up according to stimulus approaches**

prominent issue with electrical stimulations: 
+ each stimulus may be generated by a different device
+ each stimulus is delivered through a electrode contact and not the electrode / electrode array device

Note : specifically the last point is not solved consistently yet. Schema model needs further discussion for this.

**Comment by lzehl** [comment]
use cases for pure electrical stimulations that may help the discussion:

**A)** Single Electrode; one Cell; one Stimulus repeated 100 times
**B)** Single Electrode; one Cell; three Stimuli; each repeated 100 times; composition conditions: random (with or without delay between stimuli) OR alternating (with stimulus block of 2; with or without delay between stimuli, and/or with or without delay between stimulus blocks)
**C)** Electrode Array (electrode/contact X is stimulated); cell population; one Stimulus repeated 100 times
**D)** Electrode Array (electrode/contact X is stimulated); cell population; three Stimuli; each repeated 100 times; conditions: random OR alternating (with stimulus block of 2)

**Comment by Peyman-N** [comment]
**note for future extensions**, stimulus or stimulation activity potentiality needs a location

**Comment by lzehl** [comment]
Nearly all aspects of this issue were now solved in the openMINDS_stimulation and openMINDS_ephys extension.
@UlrikeS91 let's discuss and coordinate on the TODOs outside of the two extensions.

TODOs for openMINDS_controlledTerms:

1) separate terminology for StimulationTechnique from Technique
2) move "electrical stimulation" from Technique to StimulationApproach
3) add "optical stimulation" to StimulationApproach
4) split terminology StimulusType into approach specific types (e.g. ElectricalStimulusType)
5) separate terminology for AnalysisTechnique from Technique

TODOs for openMINDS_core:

6) change property "stimulation" in BehavioralProtocol and Protocol to link to StimulationTechnique, and StimulationApproach
7) add property "analysis" to Protocol with link to AnalysisTechnique
8) update links in "technique" under DSV to all techniques terminologies (Technique, AnalysisTechnique, StimulationTechnique)
8) delete Stimulation schema from core
